x-pack/auditbeat/processors/sessionmd/provider/modernprovider/modernprovider_linux.go

```diff
@@ -0,0 +1,523 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux && (amd64 || arm64) && cgo
+
+package modernprovider
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/user"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	quark "github.com/elastic/go-quark"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/sessionmd/provider"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/sessionmd/types"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type prvdr struct {
+	ctx    context.Context
+	logger *logp.Logger
+	qq     *quark.Queue
+	qqMtx  *sync.Mutex
+}
+
+type TTYType int
+
+const (
+	TTYUnknown TTYType = iota
+	Pts
+	TTY
+	TTYConsole
+)
+
+type EntryType string
+
+const (
+	Init         EntryType = "init"
+	Sshd         EntryType = "sshd"
+	Ssm          EntryType = "ssm"
+	Container    EntryType = "container"
+	Terminal     EntryType = "terminal"
+	EntryConsole EntryType = "console"
+	EntryUnknown EntryType = "unknown"
+)
+
+const (
+	ptsMinMajor     = 136
+	ptsMaxMajor     = 143
+	ttyMajor        = 4
+	consoleMaxMinor = 63
+	ttyMaxMinor     = 255
+	retryCount      = 2
+)
+
+var (
+	bootID     string
+	pidNsInode uint64
+)
+
+func readBootID() (string, error) {
+	bootID, err := os.ReadFile("/proc/sys/kernel/random/boot_id")
+	if err != nil {
+		panic(fmt.Sprintf("could not read /proc/sys/kernel/random/boot_id: %v", err))
+	}
+
+	return strings.TrimRight(string(bootID), "\n"), nil
+}
+
+func readPIDNsInode() (uint64, error) {
+	var ret uint64
+
+	pidNsInodeRaw, err := os.Readlink("/proc/self/ns/pid")
+	if err != nil {
+		panic(fmt.Sprintf("could not read /proc/self/ns/pid: %v", err))
+	}
+
+	if _, err = fmt.Sscanf(pidNsInodeRaw, "pid:[%d]", &ret); err != nil {
+		panic(fmt.Sprintf("could not parse contents of /proc/self/ns/pid (%s): %v", pidNsInodeRaw, err))
+	}
+
+	return ret, nil
+}
+
+func NewProvider(ctx context.Context, logger *logp.Logger) (provider.Provider, error) {
+
+	attr := quark.DefaultQueueAttr()
+	attr.Flags = quark.QQ_KPROBE | quark.QQ_MIN_AGG | quark.QQ_ENTRY_LEADER
```

Comment:
```suggestion
	attr.Flags = quark.QQ_ENTRY_LEADER
```

I'm just getting my war through this diff, but if I understood the comments right, you want to eat the old EBPF implementation, so by *not* specifying QQ_KPROBE, quark will try ebpf, then kprobe, if you *do* specify QQ_KPROBE, it will *not* try EBPF.

As I noted in another comment, if you're just using lookups on the queue (which is perfectly fine), there is no reason to use `QQ_MIN_AGG`, the hold-off time will be the same, and MIN_AGG will just end up generating more events that you're not interested in.